scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala

```diff
@@ -813,9 +849,29 @@ class ScalametaParser(input: Input)(implicit dialect: Dialect) { parser =>
       case _ => term
     }
 
-  private def dropOuterBlock(term: Term.Block): Term =
+  private def dropOuterBlock(term: Term.Block): Term = {
+    def commentPosOpt(pos: Int) = scannerTokens(pos) match {
+      case _: Comment => Some(pos)
+      case _: LF | _: FF if scannerTokens(pos - 1).is[Comment] => Some(pos - 1)
+      case _ => None
+    }
+    term.stats match {
+      case (b: Term.Block) :: Nil => keepInnerBlock(b)
+      case (s: Term) :: Nil if commentPosOpt(in.prevTokenPos).forall { commentPos =>
+            s.origin match {
+              case Origin.Parsed(_, _, pos) => pos.end > commentPos
+              case _ => false
+            }
+          } =>
+        s
```

Comment:
Could you explain why it's needed?
It took me a bit but as I got the logic is the following: `keep Block in case if consist from single term in case if there is a trailing comment`. Is it for scalafmt?

Also, I've tried the following case:
```scala
object y {
 val a =
   if cond then
     fx // c1
   else
      gx
  /* c2 */}
```
Having this logic `fx //c1` is parsed as a `Term.Block` while it should be a `Term.Name`.

Reply:
would you only keep the outer block if the comment was standalone? generally, i would prefer not to drop block at all, even if for a single statement, but for maximum backwards compatibility...

the original issue, #2516, had a problem with a trailing comment, not standalone.

Reply:
if we don't put block around it, then Name will include the comment.

and more on why this shouldn't be just Name. `then` expects a block or at least an expression, not a specific identifier or constant; to me, it's similar to the difference between "number" (expression) and "digit" (basic component of a number).

Reply:
I'm still not sure about this change.
The comment says ` if block ends in a comment and single term doesn't contain it, keep block` but it's incorrect.

If you revert this part and run `Scala3PositionSuite` you will notice that single terms includes comment positions.

For example:
```scala
object y {
   val a =
      fx
      // c1
}
// positions with this change
    """|Template {
       |  val a =
       |    fx
       |    // c1
       |}
       |Self   @@val a =
       |Defn.Val val a =
       |    fx
       |    // c1
       |Term.Block fx
       |    // c1
       |""".stripMargin


// positions without this change
// c1 is included in fx-term

    """|Template {
       |  val a =
       |    fx
       |    // c1
       |}
       |Self   @@val a =
       |Defn.Val val a =
       |    fx
       |    // c1
       |Term.Name fx
       |    // c1
       |""".stripMargin

```
So, the question is if there is a need to do that if non-block term pos anyway contains comments?

Reply:
that's exactly it. if you don't apply this change, then the comment will be part of `Term.Name` -- but it shouldn't be. and i am not sure if that's desirable, really.

the comment will not be lost, with or without the last commit, but i am simply suggesting that the comment should be part of the block, or that we should remove the block that was there in the first place.

how's this: can we merge the 5 commits which add comment handling, and then have a separate discussion on whether to keep or remove the so-called trivial blocks?